pkg/retriever/bitswapretriever.go

```diff
@@ -150,6 +163,42 @@ func (br *bitswapRetrieval) RetrieveFromAsyncCandidates(ayncCandidates types.Inb
 	}
 	br.events(events.Started(br.request.RetrievalID, phaseStartTime, types.RetrievalPhase, bitswapCandidate))
 
+	loader := loaderForSession(br.request.RetrievalID, br.inProgressCids, br.bsGetter)
```

Comment:
seems a bit odd to have all this code between reading nextCandidates and using it? I wonder if https://github.com/filecoin-project/lassie/pull/149/files#diff-f0e11ea724fe0ea2477cdd34da7a8ca6c16a37cf78c9167d79c02c37b627e3f5R203-R217 should stay up here